Issue10

amakalev opened the pull request:
It was created method, that handle photo request from front-end. This method gets last photo in session and recognition results of this photo from data base and then sends all data to front-end. If user doesn't have any sessions, send "bad request" response, if recognition results for this photo isn't prepared, value of "recognitionResults" key will be null.

green124b-dev committed:
It was created Photo Request Model

green124b-dev committed:
Created method accepting photo request from frontend

green124b-dev committed:
Merge branch 'master' of https://github.com/alex-ks/backend-comptech into issue10

green124b-dev committed:
Added check of availability of recognition results

green124b-dev committed:
Added check of availability sessions for user

green124b-dev committed:
Added interface IAnalitycsClient and class AnalyticsClient

DVEfremov commented on Comptech.Backend/src/Comptech.Backend.Service/Controllers/PhotoController.cs:
```diff
@@ -1,12 +1,88 @@
+                    var recognitionResults = recognitionResultsRepository.GetRecognitionResultsByPhotoId(photo.PhotoID);
+                    
+                    var recRes = (null == recognitionResults) 
+                    ?
```
[codestyle] ?

amakalev replied:
в данном участке кода обрабатывается момент, что фото в базе лежит, но нахождение лица на этом фото еще не завершилось, стиль оформления тернарного оператора был вынужденым из-за большой штуки
![screenshot from 2017-02-06 18-06-06](https://cloud.githubusercontent.com/assets/22714879/22645286/df8c4e5a-ec98-11e6-93fd-b3d305ec4e53.png)

DVEfremov commented on Comptech.Backend/src/Comptech.Backend.Service/Controllers/PhotoController.cs:
```diff
@@ -1,12 +1,88 @@
+            using (logger.BeginScope(nameof(GetPhoto)))
+            {
+                logger.LogInformation("Frontend tries to get photo");
+                try {
```
лучше использовть подход при котором все предусловия (проерка корректности  переданных параметров)
проверяются в самом начале

alex-ks replied:
\+ лучше эти проверки вынести в отдельный метод

amakalev replied:
Параметры метода инстанцируются с помощью DI, потому они всегда валидны

green124b-dev committed:
It was added models for recognitioin results rescponce and added communication with server

green124b-dev committed:
Merge branch 'master' of https://github.com/alex-ks/backend-comptech into issue21

green124b-dev committed:
It was fixed bug connected with incompatibility of dependencies(Microsoft.AspNet.WebApi.Client)

green124b-dev committed:
Merge branch 'master' of https://github.com/alex-ks/backend-comptech into issue21

green124b-dev committed:
Created model for recognition request response

green124b-dev committed:
Merge branch 'master' of https://github.com/alex-ks/backend-comptech into issue21

DVEfremov commented:
Устно обсудили 
тут вроде норм все (предулосвия в том виде в котором я из предполагал не нужны, необходимые  проверки есть  за пределами этого метода )

green124b-dev committed:
Rename recogntition result response variable

green124b-dev committed:
Merge branch 'master' of https://github.com/alex-ks/backend-comptech into issue21

green124b-dev committed:
Added class for testing AnalyticsClient

green124b-dev committed:
Fixed typo in return type of Analytics client method(UploadPhoto)

green124b-dev committed:
Fixed handling of bad request response in AnalyticsClient methods

alex-ks commented:
Предлагаю добавить немного тестов.

green124b-dev committed:
Added converting photo to Base64 for transfering

green124b-dev committed:
Added integration testing for AnalyticsAgent

eliseev-eag committed:
Merge branch 'master' of https://github.com/alex-ks/backend-comptech into 'issue17'

green124b-dev committed:
Merge branch 'master' of https://github.com/alex-ks/backend-comptech into issue10

eliseev-eag committed:
added AnalyticsAgent
wrote first version of function which take photo from queue and send to analyticsClient

green124b-dev committed:
Created PhotoControllerTest

eliseev-eag committed:
Merge branch 'issue21' of https://github.com/alex-ks/backend-comptech into 'issue17'

eliseev-eag committed:
Added PollingTask, CancelPolling and Dispose methods on AnalyticsAgent
Added public methods to AnalyticsAgent which provides API to start task to recognition photo and cancel this task

green124b-dev committed:
Fixed typos

green124b-dev committed:
Added testing method

green124b-dev committed:
Added checking the validity of the answer

green124b-dev committed:
Merge branch 'master' of https://github.com/alex-ks/backend-comptech into issue21

green124b-dev committed:
It was improved code style

green124b-dev committed:
Fixed typo

green124b-dev committed:
Merge branch 'issue21' of https://github.com/alex-ks/backend-comptech into issue21

amakalev commented:
тесты добавлены

eliseev-eag committed:
fixed infinite loop when photo uploading
added param PhotoUploadTryCount - number of attempts to upload photo
rename isPhotoAploaded variable to isPhotoUploaded

green124b-dev committed:
Renamed PhotoControllerTest to GetPhotoTest

green124b-dev committed:
Fixed typos im urls of requests

green124b-dev committed:
Merge branch 'issue21' of https://github.com/alex-ks/backend-comptech into issue21

green124b-dev committed:
It was fixed bug in deserialization

alex-ks committed:
Merge pull request #34 from alex-ks/issue21

Implemented AnalyticsClient

green124b-dev committed:
Merge branch 'master' of https://github.com/alex-ks/backend-comptech into issue10
